stub-runner/stub-runner-spring/src/main/groovy/com/ofg/stub/StubDownloader.groovy

```diff
@@ -0,0 +1,157 @@
+    }
+
+    private setDependencyResolver(boolean skipLocalRepo) {
+        if (this.dependencyResolver == null) {
+            this.dependencyResolver = buildResolver(skipLocalRepo)
+        }
+
+    }
+
+    private DependencyResolver buildResolver(boolean skipLocalRepo) {
+        skipLocalRepo ? new RemoteDependencyResolver() : new LocalFirstDependencyResolver()
+    }
+
+    /**
+     * Dependency resolver providing {@link URI} to remote dependencies.
+     */
+    @Slf4j
+    private class RemoteDependencyResolver extends DependencyResolver {
+
+        URI resolveDependency(String stubRepositoryRoot, Map depToGrab) {
+            try {
+                return doResolveRemoteDependency(stubRepositoryRoot, depToGrab)
+            }
+            catch (UnknownHostException e) {
+                failureHandler(stubRepositoryRoot, "unknown host error -> ${e.message}", e)
+            }
+            catch (Exception e) {
+                failureHandler(stubRepositoryRoot, "connection error -> ${e.message}", e)
+            }
+        }
+
+        private URI doResolveRemoteDependency(String stubRepositoryRoot, Map depToGrab) {
+            addResolver(name: REPOSITORY_NAME, root: stubRepositoryRoot)
+            log.info("Resolving dependency ${depToGrab} location in remote repository...")
+            URI resolvedUri = resolveDependencyLocation(depToGrab)
+            ensureThatLatestVersionWillBePicked(resolvedUri)
+            return resolveDependencyLocation(depToGrab)
+        }
+
+        private void failureHandler(String stubRepository, String reason, Exception cause) {
+            throw new DependencyResolutionException("Unable to open connection with stub repository [$stubRepository]. Reason: $reason", cause)
+        }
+
+        private void ensureThatLatestVersionWillBePicked(URI resolvedUri) {
+            getStubRepositoryGrapeRoot(resolvedUri).eachFileRecurse(FILES, {
+                if (it.name.endsWith('xml')) {
+                    log.info("removing ${it}"); it.delete()
+                }
+            })
+        }
+
+        private File getStubRepositoryGrapeRoot(URI resolvedUri) {
+            return new File(resolvedUri).parentFile.parentFile
+        }
+
+    }
+
+    /**
+     * Dependency resolver that first checks if a dependency is available in the local repository.
+     * If not, it will try to provide {@link URI} from the remote repository.
+     *
+     * @see RemoteDependencyResolver
+     */
+    @Slf4j
+    private class LocalFirstDependencyResolver extends DependencyResolver {
+
+        private DependencyResolver delegate = new RemoteDependencyResolver()
+
+        URI resolveDependency(String stubRepositoryRoot, Map depToGrab) {
+            log.info("Resolving dependency ${depToGrab} location in local repository...")
+            try {
+                resolveDependencyLocation(depToGrab)
+            } catch (Exception ignored) {
```

Comment:
Maybe we could distinguish between "normal" exception from Grape/Ivy that dependency wasn't resolve and other "unexpected" exceptions which could be logged with stacktrace?
Btw, in `Grape.java` there is:
```
    public static URI[] resolve(Map<String, Object> args, List depsInfo, Map... dependencies) {
        URI[] uris = null;
        if (enableGrapes) {
            GrapeEngine instance = getInstance();
            if (instance != null) {
                if (!args.containsKey(AUTO_DOWNLOAD_SETTING)) {
                    args.put(AUTO_DOWNLOAD_SETTING, enableAutoDownload);
                }
                if (!args.containsKey(DISABLE_CHECKSUMS_SETTING)) {
                    args.put(DISABLE_CHECKSUMS_SETTING, disableChecksums);
                }
                uris = instance.resolve(args, depsInfo, dependencies);
            }
        }
        if (uris == null) {
            return new URI[0];
        } else {
            return uris;
        }
    }
```
Is it possible to gen empty collection instead of an exception on resolution failure?